### ISSUE TITLE
Reaching download page via HTTP 301 and 302 redirection will not lead to correct plugin (stuck on DefaultPlugin)

### ISSUE THREAD

**Issue by JayFoxRox** [pyload/pyload#4193]
If I'm adding a link which redirects (through HTTP status 301 or 302) to pyload, it will fail to apply the wanted plugin which could handle the target location after the redirect.

Example: I've created a redirection at https://the-redirector.senorita.workers.dev/sandstorm which returns HTTP 301 (Moved permanently).

- When adding the redirection URL (https://the-redirector.senorita.workers.dev/sandstorm) to pyload, it will follow the redirect, but start downloading the youtube page using DefaultPlugin, instead of downloading the video. This probably isn't wanted.
- When adding the actual youtube URL (https://www.youtube.com/watch?v=y6120QOlsfU) to pyload, it will successfully download the video using the YoutubeCom plugin.

So while pyload has the concept of link decryptors, the most basic form of URL obfuscation does not seem to be supported at the moment.

I'm not sure if this is intentional, but I'd consider it a bug, not a feature.

**Comment by GammaC0de** [comment]
If you want to make pyload support `the-redirector.senorita.workers.dev`, you have to write a decrypter plugin or at least ask for one.

pyload does not support URL redirection out of the box.

**Comment by JayFoxRox** [comment]
Right, I only set up redirection on http://the-redirector.senorita.workers.dev as an example. My real URLs are getting redirected from elsewhere.

What I'm asking is support for URL redirection support in DefaultPlugin, regardless of the URL.

If the URL of the download changes using DefaultPlugin, then the URL pattern matching for plugins should happen again (= make DefaultPlugin behave like a decypter for redirects).

If this isn't a bug (why not?) then consider it a feature-request.
